crates/cli-support/src/lib.rs

```diff
@@ -380,3 +380,28 @@ impl fmt::Display for MyError {
         self.0.fmt(f)
     }
 }
+
+fn reset_indentation(s: &str) -> String {
+    indent_recurse(s.lines(), 0)
+}
+
+fn indent_recurse(mut lines: ::std::str::Lines, current_indent: usize) -> String {
+    if let Some(line) = lines.next() {
+        let mut trimmed = line.trim().to_owned();
+        let mut next_indent = current_indent;
+        let mut current_indent = current_indent;
+        if trimmed.ends_with('{') {
+            next_indent += 1;
+        }
+        if trimmed.starts_with('}') || trimmed.ends_with('}') {
+            if current_indent > 0 { current_indent -= 1; }
```

Comment:
Nitpick: we shouldn't put `if` blocks on the same line as their condition:

```rust
if current_index > 0 {
    current_index -= 1;
}
```

Reply:
fixed!